pandasql/io.py

```diff
@@ -17,3 +40,52 @@ def read_numpy(*args, name=None, **kwargs):
 
 def read_pickle(*args, name=None, **kwargs):
     return DataFrame(pd.read_pickle(*args, **kwargs), name=name)
+
+
+def _csv_to_sql(*args, name, **kwargs):
+    kwargs['nrows'] = SAMPLE_LINES
+    chunk = pd.read_csv(*args, **kwargs)
+    chunk.to_sql(name=name, con=SQL_CON, index=False, if_exists='append')
+    subprocess.call(["sqlite3", DB_FILE, ".mode csv",
+                     f".import  \'| tail -n +{SAMPLE_LINES + 2} {args[0]}\' {name}"])   # noqa
+
+    df = pd.DataFrame(columns=chunk.columns)
+    return DataFrame(df, name=name, offload=False, loaded_on_sqlite=True)
```

Comment:
(Also see my second comment above) We should not be passing in a Pandas DataFrame here, because that will result in an incorrect value being stored in the returned object's `_cached_result`.

If I understand correctly, you're doing this so that the columns of the returned object are set correctly. Probably better to do this explicitly, either by setting the return value's `columns` here, or with a new `columns` parameter in `DataFrame.__init__`.